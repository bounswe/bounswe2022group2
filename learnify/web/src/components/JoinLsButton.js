import * as React from 'react';
import { useEffect } from 'react';
import './component_styles.css';

export default function JoinLsButton(props) {
    const [isClicked, setIsClicked] = React.useState(false);
    const token = localStorage.getItem('token');
    const username = localStorage.getItem('username');
    const isJoined = true ? props.participants.some((participant) => participant === username) : false;


    const joinLs = async () => {
        setIsClicked(true);
        const response = await fetch(`${process.env.REACT_APP_BACKEND_BASE_URL}learningspace/enroll`, {
            method: 'POST',
            body: JSON.stringify({
                title: props.title,
            }),
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `${token}`
            }
        });
        window.location.reload();
        /*
        const data = await response.json();
        if (data.success) {
            setIsJoined(true);
        }*/
    }



    return (
        <div>
            {isJoined ? <button className="btn-white2" disabled>Joined</button> : <button className="btn-orange" onClick={joinLs}>Join</button>}
            
        </div>
    );
}

// <button className={isClicked ? "btn-white2" :"btn-orange"} onClick={joinLs} >{isClicked ? 'Joined': 'Join'}</button>