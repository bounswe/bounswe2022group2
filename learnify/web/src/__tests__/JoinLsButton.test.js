import React from "react";
import JoinLsButton from "./../components/JoinLsButton";
import { screen, render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe("JoinLsButton", () => {
    test('should render component layout correctly', async () => {
        const component = renderer.create(<BrowserRouter><JoinLsButton participants = {[ "gokay", "koray", "altay", "enes"]}/></BrowserRouter>).toJSON();
        expect(component).toMatchSnapshot();
    });
}   );

test('join ls button renders correctly', async () => {
    render(<BrowserRouter><JoinLsButton participants = {[ "gokay", "koray", "altay", "enes"]}/></BrowserRouter>);
    const button = screen.getByText(/Join/i);
    expect(button).toBeInTheDocument();
}  );
