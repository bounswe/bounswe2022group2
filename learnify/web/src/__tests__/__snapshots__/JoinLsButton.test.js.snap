// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinLsButton should render component layout correctly 1`] = `
<div>
  <button
    className="btn-orange"
    onClick={[Function]}
  >
    Join
  </button>
</div>
`;
